src/nrtk_explorer/library/object_detector.py

```diff
@@ -69,10 +67,17 @@ def eval(
                 img = content[path]
             else:
                 img = self.manager.load_image(path)
+            images.setdefault(img.size, {})[path] = img
 
-            images.setdefault(img.size, []).append(img)
+        def run_group(group):
+            imgs = list(group.values())
+            predictions = self.pipeline(imgs, batch_size=batch_size)
+            # { path -> prediction }
+            paths = group.keys()
+            return {path: pred for path, pred in zip(paths, predictions)}
 
-        # Call by each group
-        predictions = [self.pipeline(group, batch_size=batch_size) for group in images.values()]
```

Comment:
@vicentebolea Why do we want to run the pipeline in groups of same shaped images?  Performance/memory optimization?